test/elixir/test/delayed_commits_test.exs

```diff
@@ -0,0 +1,31 @@
+defmodule DelayedCommitsTest do
+  use CouchTestCase
+
+  @moduledoc """
+  Test CouchDB delayed commits
+  This is a port of the delayed_commits.js suite
+
+  Note that delayed_commits is deprecated in 2.0, so this is a minimal
+  test to show it still works. delayed_commits will be removed in 3.0.
+  """
+
+  @tag config: [
+         {"couchdb", "delayed_commits", "true"}
+       ]
+  @tag :with_db
+  test "delayed commit", context do
```

Comment:
The implementation of delayed commits is prune to race conditions since it is implemented using send_after see https://github.com/apache/couchdb/blob/master/src/couch/src/couch_db_updater.erl#L742. The timeout is only one second. Which makes the test a flaky one. I would recommend wrapping all steps with `retry_until(fn -> ... true end)`

Reply:
There is a slight problem with using `retry_until`. Most likely it would swallow the exception when timeout is reached so we wouldn't see nice error message. Maybe we need to talk to @jaydoane to extend it to re-raise original error.

Reply:
This test is racy pretty much by definition, though most of the problems with its javascript version coming not from the fact that it takes more than one second to _send_ a node to a restart, but that the cluster doesn't come back in time.

I'm honestly sceptical about usage of `retry_until` here. It has default timeout at 5 secs and this test takes whooping 9 sec on my MBP with the whole cluster bounce. I can wrap it with something like 10 sec retry interval and total timeout 30 secs, but this is pretty big band aid, so I'd prefer to actually see that race where doc saved before cluster bounce before extending the suites' runtime that much.